pkg/deploy/lattice/listener_synthesizer.go

```diff
@@ -80,6 +89,45 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
 	return nil
 }
 
+func (l *listenerSynthesizer) getLatticeListenerDefaultAction(ctx context.Context, stackListener *model.Listener) (
+	*vpclattice.RuleAction, error,
+) {
+	if stackListener.Spec.DefaultAction == nil ||
+		(stackListener.Spec.DefaultAction.FixedResponseStatusCode == nil && stackListener.Spec.DefaultAction.Forward == nil) {
+		return nil, fmt.Errorf("invalid listener default action, must be either fixed response or forward")
+	}
```

Comment:
We should rather validate this when we add the listener to the stack in `listener.go#NewListener()`.

Reply:
Will mimic the `func (t *TargetGroupSpec) Validate()`, and put that validation in the new function `func (spec *ListenerSpec) Validate()`